diff --git a/ports/atk/portfile.cmake b/ports/atk/portfile.cmake
index 807f5674ce..994a57b1c4 100644
--- a/ports/atk/portfile.cmake
+++ b/ports/atk/portfile.cmake
@@ -1,10 +1,8 @@
-vcpkg_from_gitlab(
-    GITLAB_URL https://gitlab.gnome.org/
+vcpkg_from_git(
+    URL https://gitlab.gnome.org/Archive/atk.git
     OUT_SOURCE_PATH SOURCE_PATH
-    REPO GNOME/atk
-    REF "${VERSION}"
-    HEAD_REF master
-    SHA512 f31951ecbdace6a18fb9f772616137cb8732163b37448fef4daf1af60ba8479c94d498dcdaf4880468c80012c77a446da585926a99704a9a940b80e546080cf3
+    REF "f1051ba57a2110c46a136a48f4d60a405bc2c3f5"
+    HEAD_REF "${VERSION}"
 )
 
 if("introspection" IN_LIST FEATURES)
diff --git a/ports/gdk-pixbuf/portfile.cmake b/ports/gdk-pixbuf/portfile.cmake
index 82d534b447..8b7479a787 100644
--- a/ports/gdk-pixbuf/portfile.cmake
+++ b/ports/gdk-pixbuf/portfile.cmake
@@ -1,10 +1,8 @@
-vcpkg_from_gitlab(
-    GITLAB_URL https://gitlab.gnome.org/
+vcpkg_from_git(
+    URL https://gitlab.gnome.org/GNOME/gdk-pixbuf.git
     OUT_SOURCE_PATH SOURCE_PATH
-    REPO GNOME/gdk-pixbuf
-    REF "${VERSION}"
-    SHA512 f95c92974ed6efac9845790ef5c4ed74dd6e28b182ea3732013c46b016166e92f8bc10c1994358d79ff53e988c615c43cb1a2130c6ef531ef9d84c2fdcc87e52
-    HEAD_REF master
+    REF "e4315fb8553776e13d39e3f2e0ea8792db61720c"
+    HEAD_REF "${VERSION}"
     PATCHES
         fix_build_error_windows.patch
         loaders-cache.patch
diff --git a/ports/gtk3/portfile.cmake b/ports/gtk3/portfile.cmake
index 13f2f08889..67a837e11c 100644
--- a/ports/gtk3/portfile.cmake
+++ b/ports/gtk3/portfile.cmake
@@ -6,14 +6,12 @@ if(buildtrees_path_length GREATER warning_length AND CMAKE_HOST_WIN32)
     )
 endif()
 
-vcpkg_from_gitlab(
+vcpkg_from_git(
     OUT_SOURCE_PATH SOURCE_PATH
-    GITLAB_URL https://gitlab.gnome.org
-    REPO GNOME/gtk
-    REF "${VERSION}"
-    SHA512 19e5482e4e843aa946ab79c8ce283a7b44aaac43ad99b6913cbc3c91492bf722ebe0238457b75b82be6d6c65a394d32ebc8732832f3f800145e3cf69d5c1e77c
+    URL https://gitlab.gnome.org/GNOME/gtk
+    REF "0d3e55cf22abc81b532a3011462b3195f1480974"
+    HEAD_REF "3.24.51"
     PATCHES
-        0001-build.patch
         cairo-cpp-linkage.patch
 )
 
