diff --git a/ports/atk/portfile.cmake b/ports/atk/portfile.cmake
index 807f5674ce..994a57b1c4 100644
--- a/ports/atk/portfile.cmake
+++ b/ports/atk/portfile.cmake
@@ -1,10 +1,8 @@
-vcpkg_from_gitlab(
-    GITLAB_URL https://gitlab.gnome.org/
+vcpkg_from_git(
+    URL https://gitlab.gnome.org/Archive/atk.git
     OUT_SOURCE_PATH SOURCE_PATH
-    REPO GNOME/atk
-    REF "${VERSION}"
-    HEAD_REF master
-    SHA512 f31951ecbdace6a18fb9f772616137cb8732163b37448fef4daf1af60ba8479c94d498dcdaf4880468c80012c77a446da585926a99704a9a940b80e546080cf3
+    REF "f1051ba57a2110c46a136a48f4d60a405bc2c3f5"
+    HEAD_REF "${VERSION}"
 )
 
 if("introspection" IN_LIST FEATURES)
diff --git a/ports/gdk-pixbuf/portfile.cmake b/ports/gdk-pixbuf/portfile.cmake
index 82d534b447..8b7479a787 100644
--- a/ports/gdk-pixbuf/portfile.cmake
+++ b/ports/gdk-pixbuf/portfile.cmake
@@ -1,10 +1,8 @@
-vcpkg_from_gitlab(
-    GITLAB_URL https://gitlab.gnome.org/
+vcpkg_from_git(
+    URL https://gitlab.gnome.org/GNOME/gdk-pixbuf.git
     OUT_SOURCE_PATH SOURCE_PATH
-    REPO GNOME/gdk-pixbuf
-    REF "${VERSION}"
-    SHA512 f95c92974ed6efac9845790ef5c4ed74dd6e28b182ea3732013c46b016166e92f8bc10c1994358d79ff53e988c615c43cb1a2130c6ef531ef9d84c2fdcc87e52
-    HEAD_REF master
+    REF "e4315fb8553776e13d39e3f2e0ea8792db61720c"
+    HEAD_REF "${VERSION}"
     PATCHES
         fix_build_error_windows.patch
         loaders-cache.patch
diff --git a/ports/gtk3/cairo-cpp-linkage.patch b/ports/gtk3/cairo-cpp-linkage.patch
index dce7739b62..b699c3f451 100644
--- a/ports/gtk3/cairo-cpp-linkage.patch
+++ b/ports/gtk3/cairo-cpp-linkage.patch
@@ -1,8 +1,8 @@
 diff --git a/gtk/meson.build b/gtk/meson.build
-index 79f4f0d..33a7723 100644
+index ea866d8231..0d312f3e0c 100644
 --- a/gtk/meson.build
 +++ b/gtk/meson.build
-@@ -1074,6 +1074,7 @@ gtk_query_settings = executable(
+@@ -1102,6 +1102,7 @@ gtk_query_settings = executable(
    'gtk-query-settings.c',
    c_args: gtk_cargs,
    dependencies: libgtk_dep,
@@ -10,7 +10,7 @@ index 79f4f0d..33a7723 100644
    install: true
  )
  gtk_tools += gtk_query_settings
-@@ -1083,6 +1084,7 @@ gtk_builder_tool = executable(
+@@ -1111,6 +1112,7 @@ gtk_builder_tool = executable(
    'gtk-builder-tool.c',
    c_args: gtk_cargs,
    dependencies: libgtk_dep,
@@ -18,7 +18,7 @@ index 79f4f0d..33a7723 100644
    install: true
  )
  gtk_tools += gtk_builder_tool
-@@ -1115,6 +1117,7 @@ gtk_update_icon_cache = executable(
+@@ -1143,6 +1145,7 @@ gtk_update_icon_cache = executable(
    extra_update_icon_cache_objs,
    c_args: gtk_cargs,
    dependencies: libgtk_dep,
@@ -26,7 +26,7 @@ index 79f4f0d..33a7723 100644
    install: true
  )
  gtk_tools += gtk_update_icon_cache
-@@ -1125,6 +1128,7 @@ gtk_query_immodules = executable(
+@@ -1153,6 +1156,7 @@ gtk_query_immodules = executable(
    'gtkutils.c',
    c_args: gtk_cargs,
    dependencies: libgtk_dep,
@@ -34,7 +34,7 @@ index 79f4f0d..33a7723 100644
    install: true
  )
  gtk_tools += gtk_query_immodules
-@@ -1134,6 +1138,7 @@ gtk_encode_symbolic_svg = executable(
+@@ -1162,6 +1166,7 @@ gtk_encode_symbolic_svg = executable(
    'encodesymbolic.c',
    c_args: gtk_cargs,
    dependencies: libgtk_dep,
@@ -42,7 +42,7 @@ index 79f4f0d..33a7723 100644
    install: true
  )
  gtk_tools += gtk_encode_symbolic_svg
-@@ -1143,6 +1148,7 @@ gtk_launch = executable(
+@@ -1171,6 +1176,7 @@ gtk_launch = executable(
    'gtk-launch.c',
    c_args: gtk_cargs,
    dependencies: libgtk_dep,
@@ -51,12 +51,12 @@ index 79f4f0d..33a7723 100644
  )
  gtk_tools += gtk_launch
 diff --git a/meson.build b/meson.build
-index 287f0cb..d35106f 100644
+index 397ea0730b..030e67806d 100644
 --- a/meson.build
 +++ b/meson.build
 @@ -1,4 +1,4 @@
--project('gtk+', 'c',
-+project('gtk+', 'c', 'cpp',
-   version: '3.24.43',
+-project('gtk', 'c',
++project('gtk', 'c', 'cpp',
+   version: '3.24.51',
    default_options: [
      'buildtype=debugoptimized',
diff --git a/ports/gtk3/portfile.cmake b/ports/gtk3/portfile.cmake
index 13f2f08889..3dc3a11553 100644
--- a/ports/gtk3/portfile.cmake
+++ b/ports/gtk3/portfile.cmake
@@ -6,12 +6,11 @@ if(buildtrees_path_length GREATER warning_length AND CMAKE_HOST_WIN32)
     )
 endif()
 
-vcpkg_from_gitlab(
+vcpkg_from_git(
     OUT_SOURCE_PATH SOURCE_PATH
-    GITLAB_URL https://gitlab.gnome.org
-    REPO GNOME/gtk
-    REF "${VERSION}"
-    SHA512 19e5482e4e843aa946ab79c8ce283a7b44aaac43ad99b6913cbc3c91492bf722ebe0238457b75b82be6d6c65a394d32ebc8732832f3f800145e3cf69d5c1e77c
+    URL https://gitlab.gnome.org/GNOME/gtk.git
+    REF "0d3e55cf22abc81b532a3011462b3195f1480974"
+    HEAD_REF "3.24.51"
     PATCHES
         0001-build.patch
         cairo-cpp-linkage.patch
diff --git a/ports/libepoxy/portfile.cmake b/ports/libepoxy/portfile.cmake
index d39b95423d..8e04841260 100644
--- a/ports/libepoxy/portfile.cmake
+++ b/ports/libepoxy/portfile.cmake
@@ -15,7 +15,7 @@ vcpkg_from_github(
 )
 
 if (VCPKG_TARGET_IS_WINDOWS OR VCPKG_TARGET_IS_OSX)
-    set(OPTIONS -Dglx=no -Degl=no -Dx11=false)
+    set(OPTIONS -Dglx=no -Degl=yes -Dx11=false)
 elseif(VCPKG_TARGET_IS_ANDROID)
     set(OPTIONS -Dglx=no -Degl=yes -Dx11=false)
 else()
