diff --git a/ports/gdk-pixbuf/portfile.cmake b/ports/gdk-pixbuf/portfile.cmake
index 82d534b447..3545a65abc 100644
--- a/ports/gdk-pixbuf/portfile.cmake
+++ b/ports/gdk-pixbuf/portfile.cmake
@@ -3,7 +3,7 @@ vcpkg_from_gitlab(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO GNOME/gdk-pixbuf
     REF "${VERSION}"
-    SHA512 f95c92974ed6efac9845790ef5c4ed74dd6e28b182ea3732013c46b016166e92f8bc10c1994358d79ff53e988c615c43cb1a2130c6ef531ef9d84c2fdcc87e52
+    SHA512 5e499ee94e6793aab66801d6b9f3261006b335757fcd59c49aac4cbe096c9ee7d8371db8a050ddaef8b0cce561a278988a8439631533c6d8380339a4c1e90098
     HEAD_REF master
     PATCHES
         fix_build_error_windows.patch
